File: from-3401-to-3500/3438-find-valid-pair-of-adjacent-digits-in-string/solution.go
func findValidPair(s string) string {
    freq := make(map[int]int)
    n := len(s)
    nums := make([]int, n)
    for i, c := range s {
        nums[i] = int(c - '0')
        freq[nums[i]] += 1
    }
    for i := 1; i < n; i ++ {
        if nums[i - 1] != nums[i] && freq[nums[i - 1]] == nums[i - 1] && freq[nums[i]] == nums[i] {
            return s[i - 1: i + 1]
        }
    }
    return ""
}

File: from-3401-to-3500/3438-find-valid-pair-of-adjacent-digits-in-string/solution.py
class Solution:
    def findValidPair(self, s: str) -> str:
        freq = Counter(s)
        prev_c = s[0]
        prev_n = int(prev_c)
        for i in range(1, len(s)):
            curr_c = s[i]
            curr_n = int(curr_c)
            if curr_c != prev_c and freq[prev_c] == prev_n and freq[curr_c] == curr_n:
                return s[i-1:i+1]
            prev_c, prev_n = curr_c, curr_n
        return ""